src/formbuilder.tsx

```diff
@@ -287,24 +295,23 @@ export interface UseFormBuilderReturn<
  */
 export function useFormBuilder<
   TFieldValues extends FieldValues = FieldValues,
-  TContext extends object = object
+  TContext extends object = object,
 >(
-  props?: UseFormBuilderProps<TFieldValues, TContext>
+  props?: UseFormBuilderProps<TFieldValues, TContext>,
 ): UseFormBuilderReturn<TFieldValues, TContext> {
   const methods = useForm<TFieldValues, TContext>(props as never);
   const fields = useMemo(
     () => createFormBuilder<TFieldValues>(methods, []),
-    [methods.register, methods.control]
+    [methods],
```

Comment:
Are we sure this works properly and the limited number of dependencies wasn't intended to avoid some recursion problem?

Reply:
This was a change suggested by `react-hooks/exhaustive-deps`. I believe the change is correct. As it stands, the dep array is also missing `methods.setValue`, `methods.setError` and `methods.watch`. 

The `useMemo` is supposed to reduce work being done by `createFormBuilder`, not solve some recursion problem. I think the reason for this whole construct was me thinking that the return value of  `useForm` wasn't stable, even if individual methods were. But it appears that the whole thing is in fact assigned to a ref and only created once.

https://github.com/react-hook-form/react-hook-form/blob/b6bbd8a178f355632c05f7094c586773a262f78b/src/useForm.ts#L75

`formState` is an exception as it gets set to some proxy in each `useForm` call:

https://github.com/react-hook-form/react-hook-form/blob/b6bbd8a178f355632c05f7094c586773a262f78b/src/useForm.ts#L156

However, we don't consume `formState` directly in `createFormBuilder`, instead we use `useFormState`.

Instead of attempting more of these kinds of blind optimizations I'll try to follow up with further tests and see if I can get the actual number of rerenders to be as low as we can with vanilla RHF, perhaps also revisiting the stable ref topic from #1.